dora/core/common/src/main/java/alluxio/underfs/BaseUnderFileSystem.java

```diff
@@ -191,15 +194,11 @@ public boolean isSeekable() {
   @Nullable
   @Override
   public Iterator<UfsStatus> listStatusIterable(
+      // Calling this method on non s3 UFS might result in OOM because batch based fetching
+      // is not supported and this method essentially fetches all ufs status and converts it to
+      // an iterator.
       String path, ListOptions options, String startAfter, int batchSize) throws IOException {
-    // Calling this method on non s3 UFS might result in OOM because batch based fetching
-    // is not supported and this method essentially fetches all ufs status and converts it to
-    // an iterator.
-    UfsStatus[] result = listStatus(path, options);
-    if (result == null) {
-      return null;
-    }
-    return Iterators.forArray(result);
+    return new BaseUfsStatusIterator(this, path, options);
```

Comment:
The contract of `listStatusIterable` requires:

```
   * @return An iterator of ufs status. Returns
   *  {@code null} if this abstract pathname does not denote a directory.
```

So you need to check if `path` is a file, and return `null` in this case.

Reply:
Updated.